output/transform/merge_partition_sizes.go

```diff
@@ -12,11 +12,21 @@ import (
 //
 // TODO: recursively build up stats when nested partitioning is used
 func mergePartitionSizes(s snapshot.FullSnapshot, newState state.PersistedState, ts state.TransientState, databaseOidToIdx OidToIdx) snapshot.FullSnapshot {
+	relIdxToStatsIdx := make(map[int32]int, len(s.RelationStatistics))
+	for idx, stat := range s.RelationStatistics {
+		relIdxToStatsIdx[stat.RelationIdx] = idx
+	}
+
 	for idx, rel := range s.RelationInformations {
 		if !rel.HasParentRelation || rel.PartitionBoundary == "" {
 			continue
 		}
-		stat := s.RelationStatistics[idx]
+		statIdx, ok := relIdxToStatsIdx[int32(idx)]
```

Comment:
I realized that technically we should be using `int32(rel.RelationIdx)` here, instead of `idx` - in practice they are the same I think, but the way its supposed to work is that the only array you can reference by index is `RelationReferences`, and both stats and info have an explicit `RelationIdx` field that should be used to look things up.

Reply:
And we have the same problem for index statistics as well further down..